src/Config/FileConfigManager.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+namespace Spatie\Ignition\Config;
+
+use Spatie\Ignition\Contracts\ConfigManager;
+
+class FileConfigManager implements ConfigManager
+{
+    private const SETTINGS_FILE_NAME = '.ignition.json';
+
+    private string $path;
+
+    public function __construct(string $path = '')
+    {
+        $this->path = $this->initPath($path);
+    }
+
+    private function initPath(string $path): string
+    {
+        if ($path !== '') {
+            return $this->initPathFromArgument($path);
+        }
+
+        return $this->initPathFromEnvironment();
+    }
+
+    private function initPathFromArgument(string $path): string
+    {
+        if ($this->isValidPath($path)) {
+            return $path;
+        }
+
+        return '';
+    }
+
+    private function isValidPath(string $path): bool
+    {
+        if ($path === '') {
+            return false;
+        }
+
+        $path = realpath($path) ?: $path;
+
+        return file_exists($path) && is_writable($path);
+    }
+
+    protected function initPathFromEnvironment(): string
+    {
+        if ($this->isWindows()) {
+            if (empty($_SERVER['HOMEDRIVE']) || empty($_SERVER['HOMEPATH'])) {
+                return '';
+            }
+
+            $homeDirectory = $_SERVER['HOMEDRIVE'] . $_SERVER['HOMEPATH'];
+
+            return $this->preparePath($homeDirectory);
+        }
+
+        if ($homeDirectory = getenv('HOME')) {
+            return $this->preparePath($homeDirectory);
+        }
+
+        return '';
+    }
+
+    private function isWindows(): bool
+    {
+        return str_starts_with(strtoupper(PHP_OS), 'WIN');
+    }
+
+    public function load(): array
+    {
+        $file = $this->generateFullFilePath();
+
+        if (! $this->isValidFile($file)) {
+            return [];
+        }
+
+        $content = (string)file_get_contents($file);
+        $options = json_decode($content, true) ?? [];
+
+        return $options;
+    }
+
+    public function save(array $options): bool
+    {
+        $file = $this->generateFullFilePath();
+
+        if (! $this->isValidFile($file)) {
+            return false;
+        }
+
+        try {
+            file_put_contents($file, json_encode($options));
+        } catch (\Throwable) {
```

Comment:
As discussed in laravel-ignition, explicitly importing `Throwable` is more in line with Laravel's codebase